core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -209,42 +208,7 @@ public function __construct($strTable, $arrModule=array())
 			}
 		}
 
-		// Get the IDs of all root records (tree view)
-		if ($this->treeView)
-		{
-			$table = ($GLOBALS['TL_DCA'][$this->strTable]['list']['sorting']['mode'] ?? null) == 6 ? $this->ptable : $this->strTable;
-
-			// Unless there are any root records specified, use all records with parent ID 0
-			if (!isset($GLOBALS['TL_DCA'][$table]['list']['sorting']['root']) || $GLOBALS['TL_DCA'][$table]['list']['sorting']['root'] === false)
-			{
-				$objIds = $this->Database->prepare("SELECT id FROM " . $table . " WHERE pid=?" . ($this->Database->fieldExists('sorting', $table) ? ' ORDER BY sorting' : ''))
-										 ->execute(0);
-
-				if ($objIds->numRows > 0)
-				{
-					$this->root = $objIds->fetchEach('id');
-				}
-			}
-
-			// Get root records from global configuration file
-			elseif (\is_array($GLOBALS['TL_DCA'][$table]['list']['sorting']['root'] ?? null))
-			{
-				if ($GLOBALS['TL_DCA'][$table]['list']['sorting']['root'] == array(0))
-				{
-					$this->root = array(0);
-				}
-				else
-				{
-					$this->root = $this->eliminateNestedPages($GLOBALS['TL_DCA'][$table]['list']['sorting']['root'], $table, $this->Database->fieldExists('sorting', $table));
-				}
-			}
-		}
-
-		// Get the IDs of all root records (list view or parent view)
-		elseif (\is_array($GLOBALS['TL_DCA'][$this->strTable]['list']['sorting']['root'] ?? null))
-		{
-			$this->root = array_unique($GLOBALS['TL_DCA'][$this->strTable]['list']['sorting']['root']);
-		}
+		$this->initRoots();
```

Comment:
Why did you make this a separate method? It is not used anywhere else.

Reply:
Because it's nicer. We keep talking about refactoring the DC_Table so here's one step: Move the logic of deterimining the roots into its own method.